backend/webserver/views.py

```diff
@@ -69,6 +72,112 @@ def post(self, request):
         logout(request)
         return Response({'message': 'Successfully Logged out'}, status=status.HTTP_200_OK)
 
+class PostView(APIView):
+    authentication_classes = [BasicAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+    def get_post(self,post_id,author_id):
+        post = get_object_or_404(Post,id=post_id,author=author_id)
+        return post
+    
+    def get(self,request,pk,post_id, *args, **kwargs):
+        author =self.get_author(pk=pk)
+        post = self.get_post(post_id,author.id)
+        if post.visibility == "PUBLIC":
+            serializer = PostSerializer(post, context={'request': request})
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            raise Http404
+
+    def post(self, request, pk,post_id, *args, **kwargs):
+        author = self.get_author(pk=pk)
+        post = self.get_post(post_id,author.id)
+
+        if post.visibility == "PUBLIC":   
+            if author.id == request.user.id:
+                post = self.get_post(post_id,author.id)
+                date_edited = datetime.datetime.utcnow().replace(tzinfo=utc)
+                post.edited_at = date_edited
+                serializer = UpdatePostSerializer(instance=post,data=request.data,partial=True,context={'request': request})
+                if serializer.is_valid():
+                    serializer.save()
+                    return Response(serializer.data, status=status.HTTP_200_OK)
+                return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+            else:
+                return Response({'message': 'You cannot edit another authors post'}, status=status.HTTP_400_BAD_REQUEST) 
+        else:
+            return Response({'message': 'You can only edit public posts'}, status=status.HTTP_400_BAD_REQUEST)
+        
+
+    def delete(self,request,pk,post_id,*args,**kwargs):
+        author =self.get_author(pk=pk)
+        post = self.get_post(post_id,author.id)
+        if post.visibility == "PUBLIC":
+            if author.id == request.user.id:
+                post = self.get_post(post_id,author.id)
+                post.delete()
+            else:
+                return Response({'message': 'You cannot delete another authors post'}, status=status.HTTP_400_BAD_REQUEST) 
+            return Response({"message":"Object deleted!"}, status=status.HTTP_200_OK)
+        else:
+            return Response({'message': 'You can only delete public posts'}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
```suggestion
        return Response({'message': 'You can only delete public posts'}, status=status.HTTP_400_BAD_REQUEST)
```
You don't need the else clause